docs/proposals/devfile_and_default_ports_detection.md

```diff
@@ -0,0 +1,91 @@
+# Devfile port detection & default ports
+
+## Related Github Issues
+
+- https://github.com/devfile/api/issues/1154
+
+## Background
+In order to perform port detection component wise, `alizer` is performing different strategies of port detection per component. Currently, we support 3 different strategies:
+
+* Dockerfile Port Detection
+* Docker Compose Port Detection
+* Source Code Port Detection
+
+At the moment, `alizer` doesn't support port detection from `devfile.y(a)ml` files. This means that all ports defined inside a file like this will not be included in alizer's response. _As a result, the first goal of this proposal is to include the devfiles as part of the port detection algorithm_, exactly like `dockerfile` and `compose`.
+
+On the other hand, right now `alizer` is not returning any default ports of detected frameworks. As a result, for every component using a framework that has default ports, alizer will not include them in its response even if we haven't detected any ports from our strategies (ports list is empty). That said, as a second goal of this proposal, we would like to add a new feature in alizer that will include default ports in alizer's response.
+
+## Devfile Port Detection
+For devfile port detection we should follow the same flow with dockerfile:
+
+1. Create a function that will:
+    * look for a devfile inside a given component (The `utils.GetLocations` function should be updated to support devfiles too).
+    * It will also find any ports defined inside a devfile.
```

Comment:
How about clarifying this a little bit? Not sure what you mean by "ports defined inside a devfile"? Container components endpoints, regardless of exposure? Kubernetes/OpenShift components endpoints, regardless of exposure? All of them? Anything else?

Reply:
Yeap! I've updated the doc. I think we should return all ports detected under `components.kubernetes.endpoints` & `components.container.endpoints`